services/dataplatform/cluster.go

```diff
@@ -129,9 +132,48 @@ func setPatchClusterRequestProperties(d *schema.ResourceData) *dataplatform.Patc
 		dataplatformCluster.Properties.MaintenanceWindow = setMaintenanceWindowData(d)
 	}
 
+	dataplatformCluster.Properties.Lans = setLansData(d)
+
 	return &dataplatformCluster
 }
 
+func setLansData(d *schema.ResourceData) *[]dataplatform.Lan {
+	var lansBody []dataplatform.Lan
+	if lansData, ok := d.GetOk("lans"); ok {
+		if lansData, ok := lansData.(*schema.Set); ok {
+			for _, lanData := range lansData.List() {
+				if lan, ok := lanData.(map[string]interface{}); ok {
+					lanID := lan["lan_id"].(string)
+					dhcp := lan["dhcp"].(bool)
+					var lanBody dataplatform.Lan
+					lanBody.LanId = &lanID
+					lanBody.Dhcp = &dhcp
+					lanBody.Routes = setRoutesData(lan)
+					lansBody = append(lansBody, lanBody)
+				}
+			}
+		}
+	}
+	return &lansBody
+}
+
+func setRoutesData(lan map[string]interface{}) *[]dataplatform.Route {
+	var routesBody []dataplatform.Route
+	if routesData, ok := lan["routes"].(*schema.Set); ok {
```

Comment:
same

Reply:
@cristiGuranIonos same thing here, check the comment above